src/App.tsx

```diff
@@ -1,27 +1,60 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42).map(n => `Item ${n}`);
 
+function prepareItems(
+  allItems: string[],
+  itemsPerPage: number,
+  currentPage: number,
+) {
+  const start = itemsPerPage * currentPage - itemsPerPage;
+  const end =
+    start + itemsPerPage > allItems.length
+      ? allItems.length
+      : start + itemsPerPage;
+
+  return allItems.slice(start, end);
+}
+
 export const App: React.FC = () => {
+  const [itemsPerPage, setItemsPerPage] = useState(5);
+  const [currentPage, setCurrentPage] = useState(1);
+
+  const visibleItems = prepareItems(items, itemsPerPage, currentPage);
+
+  const startItem = itemsPerPage * currentPage - itemsPerPage;
+  const endItem =
+    startItem + itemsPerPage > items.length
+      ? items.length
+      : startItem + itemsPerPage;
```

Comment:
The logic for calculating `startItem` and `endItem` is duplicated here and in the `prepareItems` function. Consider removing this duplication by either reusing the `prepareItems` function here or creating a separate function that both `prepareItems` and this part of the code can use to calculate the start and end items. This will make the code DRY (Don't Repeat Yourself) and easier to maintain.